src/lib/utils.js

```diff
@@ -52,4 +53,21 @@ utils.repeatString = function (pattern/*:string*/, count/*:integer*/) {
     return result + pattern;
 };
 
+// replaces custom value tokens found in a string
+utils.replaceCustomValueTokens = function (config/*:Config*/, value/*:string*/, depth/*:integer*/)/*:string*/ {
+    depth = depth || 1;
+    if (typeof value !== 'string' || value.indexOf('#{') === -1 || !config || !config.custom) {
+        return value;
+    }
+    // Don't allow for infinite loops!
+    if (depth > 10) {
+        throw new Error('Infinite loop detected while substituting custom value tokens. Make sure your custom values don\'t contain tokens that reference one another. Aborting.');
```

Comment:
Would it be helpful to output a breadcrumb of all the variables that were traversed to get to this depth? Like $main-color -> $foo-color -> $bar-color ...

Reply:
Done!